src/Model/Behavior/DuplicatableBehavior.php

```diff
@@ -266,8 +266,10 @@ protected function _drillDownEntity($action, EntityInterface $entity, array $par
             return;
         }
 
-        foreach ($entity->{$prop} as $key => $e) {
-            $this->_drillDownEntity($action, $e, $parts, $value);
+        if (is_array($entity->{$prop}) || $entity->{$prop} instanceof \Traversable) {
+            foreach ($entity->{$prop} as $key => $e) {
```

Comment:
The $key isn't necessary

Reply:
I didn't add it, so feel that changing it is outside the scope of this PR